pandas/core/reshape/pivot.py

```diff
@@ -424,42 +425,48 @@ def _convert_by(by):
 
 @Substitution("\ndata : DataFrame")
 @Appender(_shared_docs["pivot"], indents=1)
-def pivot(data: "DataFrame", index=None, columns=None, values=None) -> "DataFrame":
+def pivot(
+    data: "DataFrame",
+    index: Optional[Union[Label, Sequence[Optional[Label]]]] = None,
+    columns: Optional[Union[Label, Sequence[Optional[Label]]]] = None,
+    values: Optional[Union[Label, Sequence[Optional[Label]]]] = None,
+) -> "DataFrame":
     if columns is None:
         raise TypeError("pivot() missing 1 required argument: 'columns'")
     columns = columns if is_list_like(columns) else [columns]
 
     if values is None:
-        cols: List[str] = []
+        cols: List[Optional[Sequence]] = []
         if index is None:
             pass
         elif is_list_like(index):
-            cols = list(index)
+            # Remove type ignore once mypy-5206 is implemented, same for below
```

Comment:
can you add a "TODO: " here?  makes it easier to track these down

Reply:
can you add an assert here to make mypy happy?